sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java

```diff
@@ -556,10 +581,233 @@ public String toJson() {
+            StoreResponseDiagnostics responseDiagnostics = resultDiagnostics.getStoreResponseDiagnostics();
+
+            String partitionId = null;
+            String[] partitionAndReplicaId = resultDiagnostics.getPartitionAndReplicaId();
+            if (partitionAndReplicaId.length == 2) {
+                partitionId = partitionAndReplicaId[0];
+            }
+
+            Collection<CosmosDiagnosticsRequestEvent> events = new ArrayList<>();
+            String pkRangeId = "";
+            double requestCharge = 0;
+            int responsePayloadLength = 0;
+            int statusCode = 0;
+            int subStatusCode = 0;
+            String activityId = requestStats.getActivityId();
+            if (responseDiagnostics != null) {
+                activityId = responseDiagnostics.getActivityId();
+                requestCharge = responseDiagnostics.getRequestCharge();
+                responsePayloadLength = responseDiagnostics.getResponsePayloadLength();
+                statusCode = responseDiagnostics.getStatusCode();
+                subStatusCode = responseDiagnostics.getSubStatusCode();
+                if (responseDiagnostics.getPartitionKeyRangeId() != null) {
+                    pkRangeId = responseDiagnostics.getPartitionKeyRangeId();
+                }
+                RequestTimeline timeline = responseDiagnostics.getRequestTimeline();
+                timeline.forEach( e -> {
+                    if (e.getStartTime() != null && e.getDuration() != null && !e.getDuration().equals(Duration.ZERO)) {
+                        events.add(new CosmosDiagnosticsRequestEvent(e.getStartTime(), e.getDuration(), e.getName()));
+                    }
+                });
+            }
+
+            Duration backendLatency = null;
+            if (resultDiagnostics.getBackendLatencyInMs() != null) {
+                backendLatency = Duration.ofNanos((long)(resultDiagnostics.getBackendLatencyInMs() * 1000000d));
+            }
+
+            CosmosDiagnosticsRequestInfo info = new CosmosDiagnosticsRequestInfo(
+                activityId,
+                partitionId,
+                pkRangeId,
+                responseStats.getRequestResourceType() + ":" + responseStats.getRequestOperationType(),
+                requestStats.getRequestStartTimeUTC(),
+                responseStats.getDuration(),
+                backendLatency,
+                requestCharge,
+                responsePayloadLength,
+                statusCode,
+                subStatusCode,
+                events
+            );
+
+            requestInfo.add(info);
+        }
+    }
+
+    private void addRequestInfoForAddressResolution(
+        ClientSideRequestStatistics requestStats,
+        ArrayList<CosmosDiagnosticsRequestInfo> requestInfo,
+        Map<String, ClientSideRequestStatistics.AddressResolutionStatistics> addressResolutionStatisticsMap
+    ) {
+        if (addressResolutionStatisticsMap == null || addressResolutionStatisticsMap.size() == 0) {
```

Comment:
Nit: `addressResolutionStatisticsMap.isEmpty()`

Reply:
Fixed in next iteration